more_itertools/more.py

```diff
@@ -741,14 +741,37 @@ def _partial(A, r):
             i += 1
             head[i:], tail[:] = tail[: r - i], tail[r - i :]
 
-    items = sorted(iterable)
+    try:
+        items = sorted(iterable)
+        sortable = True
+    except TypeError:
+        items = list(iterable)
```

Comment:
This doesn't work with an iterator, does it? The `sorted` consumes it and then `list` wont find anything. Better read it into a list first and then try `.sort()`.

Reply:
Spot on.  Indeed it doesn't.  I've fixed this now.  Shoud've checked this message first - thanks though.